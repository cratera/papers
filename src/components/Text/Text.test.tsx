import { render } from '@testing-library/react-native'

import { Text, TextProps } from '.'

import tw from '@/tailwind'

const setup = (props: TextProps) => {
  return render(<Text {...props} />)
}

describe('Text', () => {
  it('renders correctly with given props', async () => {
    const component = setup({
      children: 'Welcome to Papers!',
      variant: 'body',
      bold: true,
      style: tw`text-center`,
    }).toJSON()

    expect(component).toMatchSnapshot()
  })
})
