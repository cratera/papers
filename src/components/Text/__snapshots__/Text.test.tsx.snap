// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly with given props 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
        "fontFamily": "Karla-Bold",
        "fontSize": 16,
        "lineHeight": 24,
      },
      {
        "textAlign": "center",
      },
    ]
  }
>
  Welcome to Papers!
</Text>
`;
